File: leaf-system/src/main/java/com/leaf/service/LeafUserService.java
package com.leaf.service;

import com.leaf.annotation.Log;
import com.leaf.domain.Fetchers;
import com.leaf.domain.LeafUser;
import com.leaf.domain.LeafUserTable;
import com.leaf.record.PageRecord;
import jakarta.enterprise.context.ApplicationScoped;
import org.babyfish.jimmer.Page;
import org.babyfish.jimmer.sql.JSqlClient;

import java.util.List;

/**
 * @author jhlz
 * @version 1.0.0
 */
@ApplicationScoped
public class LeafUserService {

    private final JSqlClient sqlClient;
    LeafUserTable table = LeafUserTable.$;

    public LeafUserService(JSqlClient sqlClient) {
        this.sqlClient = sqlClient;
    }

    public Page<LeafUser> page(PageRecord page) {
        return sqlClient.createQuery(table)
                .select(table)
                .fetchPage(page.page(), page.size());
    }

    @Log(module = "详情")
    public LeafUser getById(long id) {
        return sqlClient.createQuery(table)
                .where(table.id().eq(id))
                .select(
                        table.fetch(
                                Fetchers.LEAF_USER_FETCHER
                                        .allScalarFields()
                                        .userDetail(
                                                Fetchers.LEAF_USER_DETAIL_FETCHER
                                                        .firstName()
                                                        .lastName()
                                                        .gender()
                                        )
                        )
                )
                .fetchOneOrNull();
    }

    public void add(LeafUser leafUser) {
        sqlClient.insert(leafUser);
    }

    public void update(LeafUser leafUser) {
        sqlClient.update(leafUser);
    }

    public void delete(List<Long> ids) {
        sqlClient.deleteByIds(LeafUser.class, ids);
    }
}

File: src/main/kotlin/com/jhlz/service/LeafUserService.kt
package com.jhlz.service

import com.jhlz.domain.LeafUser
import com.jhlz.domain.fetchBy
import com.jhlz.record.PageQuery
import jakarta.inject.Singleton
import org.babyfish.jimmer.Page
import org.babyfish.jimmer.sql.kt.KSqlClient

/**
 * @author jhlz
 * @version x.x.x
 */
@Singleton
class LeafUserService(val sqlClient: KSqlClient) {

    fun page(pageQuery: PageQuery): Page<LeafUser> {

        return sqlClient.createQuery(LeafUser::class) {
            select(
                table.fetchBy {
                    allScalarFields()
                    creator { username() }
                }
            )
        }.fetchPage(pageQuery.pageNum - 1, pageQuery.pageSize)
    }

    fun info(id: Long): LeafUser {
        return sqlClient.findById(LeafUser::class, id)!!
    }

    fun add(user: LeafUser) {
        sqlClient.save(user)
    }

    fun update(user: LeafUser) {
        sqlClient.save(user)
    }

    fun delete(ids: List<Long>) {
        sqlClient.deleteByIds(LeafUser::class, ids)
    }




}